Search appears to bring frontend down

Repro: No deterministic repro, but this appears to be affecting k8s.sgdev.org. Notice the multiple container restarts. If you pull the logs from the previous container, you see the following:
```
panic: runtime error: makeslice: cap out of range

goroutine 56172 [running]:
github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend.zoektIndexedRepos(0x23f77c0, 0xc010c002c0, 0xc0017701e0, 0xc002cd05a0, 0x13, 0x13, 0xc00dcee000, 0x0, 0x884f, 0x0, ...)
/buildkite/builds/buildkite-agent-6ffffdfcc4-9vdxd-1/sourcegraph/.golang/sourcegraph/src/github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend/textsearch.go:848 +0x1cc
github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend.searchFilesInRepos(0x23f77c0, 0xc010c00280, 0xc010c00240, 0x0, 0x0, 0x0, 0xc002cd05a0, 0x13, 0x13)
/buildkite/builds/buildkite-agent-6ffffdfcc4-9vdxd-1/sourcegraph/.golang/sourcegraph/src/github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend/textsearch.go:899 +0x171e
github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend.(*searchResolver).suggestFilePaths(0xc01498aae0, 0x23f7840, 0xc0134d12c0, 0x64, 0xc0134d12c0, 0xc001d4dd50, 0x0, 0x0, 0x0)
/buildkite/builds/buildkite-agent-6ffffdfcc4-9vdxd-1/sourcegraph/.golang/sourcegraph/src/github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend/search.go:571 +0x263
github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend.(*searchResolver).Suggestions.func2(0x23f7840, 0xc0134d1260, 0x0, 0x0, 0x0, 0x0, 0x0)
/buildkite/builds/buildkite-agent-6ffffdfcc4-9vdxd-1/sourcegraph/.golang/sourcegraph/src/github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend/search_suggestions.go:99 +0x255
github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend.(*searchResolver).Suggestions.func5(0xc010c16bc0, 0x23f7880, 0xc0129f1ce0, 0xc00bb288f8, 0xc013c2aac0, 0xc01498aae0, 0xc001c98eb0)
/buildkite/builds/buildkite-agent-6ffffdfcc4-9vdxd-1/sourcegraph/.golang/sourcegraph/src/github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend/search_suggestions.go:204 +0xdc
created by github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend.(*searchResolver).Suggestions
/buildkite/builds/buildkite-agent-6ffffdfcc4-9vdxd-1/sourcegraph/.golang/sourcegraph/src/github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend/search_suggestions.go:200 +0x47c
```

I saw this happen twice within 5 minutes. The query I was attempting to search for was `sourcegraph-server-gen repo:^github.com/sourcegraph/`.

--- a/cmd/frontend/graphqlbackend/textsearch.go
+++ b/cmd/frontend/graphqlbackend/textsearch.go
@@ -844,11 +844,6 @@ func zoektIndexedRepos(ctx context.Context, z *searchbackend.Zoekt, revs []*sear
 		return nil, nil, err
 	}
 
-	unindexedCap := len(revs) - len(set)
-	if unindexedCap < 0 {
-		unindexedCap = 0
-	}
-
 	indexed = make([]*search.RepositoryRevisions, 0, count)
 	unindexed = make([]*search.RepositoryRevisions, 0, len(revs)-count)
 

--- a/cmd/frontend/graphqlbackend/textsearch.go
+++ b/cmd/frontend/graphqlbackend/textsearch.go
@@ -849,8 +849,8 @@ func zoektIndexedRepos(ctx context.Context, z *searchbackend.Zoekt, revs []*sear
 		unindexedCap = 0
 	}
 
-	indexed = make([]*search.RepositoryRevisions, 0, len(set))
-	unindexed = make([]*search.RepositoryRevisions, 0, unindexedCap)
+	indexed = make([]*search.RepositoryRevisions, 0, count)
+	unindexed = make([]*search.RepositoryRevisions, 0, len(revs)-count)
 
 	for _, rev := range revs {
 		repo, ok := set[strings.ToLower(string(rev.Repo.Name))]

--- a/cmd/frontend/graphqlbackend/textsearch.go
+++ b/cmd/frontend/graphqlbackend/textsearch.go
@@ -844,8 +844,13 @@ func zoektIndexedRepos(ctx context.Context, z *searchbackend.Zoekt, revs []*sear
 		return nil, nil, err
 	}
 
+	unindexedCap := len(revs) - len(set)
+	if unindexedCap < 0 {
+		unindexedCap = 0
+	}
+
 	indexed = make([]*search.RepositoryRevisions, 0, len(set))
-	unindexed = make([]*search.RepositoryRevisions, 0, len(revs)-len(set))
+	unindexed = make([]*search.RepositoryRevisions, 0, unindexedCap)
 
 	for _, rev := range revs {
 		repo, ok := set[strings.ToLower(string(rev.Repo.Name))]

--- a/cmd/frontend/graphqlbackend/textsearch_test.go
+++ b/cmd/frontend/graphqlbackend/textsearch_test.go
@@ -686,29 +686,44 @@ func Test_zoektIndexedRepos(t *testing.T) {
 	zoekt := &searchbackend.Zoekt{Client: &fakeSearcher{repos: zoektRepoList}}
 	ctx := context.Background()
 
-	indexed, unindexed, err := zoektIndexedRepos(ctx, zoekt, repos)
-	if err != nil {
-		t.Fatal(err)
+	makeIndexed := func(repos []*search.RepositoryRevisions) []*search.RepositoryRevisions {
+		var indexed []*search.RepositoryRevisions
+		for _, r := range repos {
+			rev := *r
+			rev.IndexedHEADCommit = "deadbeef"
+			indexed = append(indexed, &rev)
+		}
+		return indexed
 	}
 
-	for _, tc := range []struct {
-		name       string
-		have, want []*search.RepositoryRevisions
-	}{
-		{"indexed", indexed, func() (want []*search.RepositoryRevisions) {
-			for i := 0; i < 3; i++ {
-				rev := *repos[i]
-				rev.IndexedHEADCommit = "deadbeef"
-				want = append(want, &rev)
+	cases := []struct {
+		name      string
+		repos     []*search.RepositoryRevisions
+		indexed   []*search.RepositoryRevisions
+		unindexed []*search.RepositoryRevisions
+	}{{
+		name:      "all",
+		repos:     repos,
+		indexed:   makeIndexed(repos[:3]),
+		unindexed: repos[3:],
+	}}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			indexed, unindexed, err := zoektIndexedRepos(ctx, zoekt, tc.repos)
+			if err != nil {
+				t.Fatal(err)
 			}
-			return
-		}()},
-		{"unindexed", unindexed, repos[3:]},
-	} {
-		if !reflect.DeepEqual(tc.have, tc.want) {
-			diff := cmp.Diff(tc.have, tc.want)
-			t.Fatalf("%s has wrong repo revs. diff=%s", tc.name, diff)
-		}
+
+			if !reflect.DeepEqual(tc.indexed, indexed) {
+				diff := cmp.Diff(tc.indexed, indexed)
+				t.Error("unexpected indexed:", diff)
+			}
+			if !reflect.DeepEqual(tc.unindexed, unindexed) {
+				diff := cmp.Diff(tc.unindexed, unindexed)
+				t.Error("unexpected unindexed:", diff)
+			}
+		})
 	}
 }
 

--- a/cmd/frontend/graphqlbackend/textsearch_test.go
+++ b/cmd/frontend/graphqlbackend/textsearch_test.go
@@ -706,6 +706,16 @@ func Test_zoektIndexedRepos(t *testing.T) {
 		repos:     repos,
 		indexed:   makeIndexed(repos[:3]),
 		unindexed: repos[3:],
+	}, {
+		name:      "one unindexed",
+		repos:     repos[3:4],
+		indexed:   repos[:0],
+		unindexed: repos[3:4],
+	}, {
+		name:      "one indexed",
+		repos:     repos[:1],
+		indexed:   makeIndexed(repos[:1]),
+		unindexed: repos[:0],
 	}}
 
 	for _, tc := range cases {

